src/node_crypto.cc

```diff
@@ -4280,8 +4280,14 @@ void CipherBase::Update(const FunctionCallbackInfo<Value>& args) {
   // Only copy the data if we have to, because it's a string
   if (args[0]->IsString()) {
     StringBytes::InlineDecoder decoder;
+    enum encoding enc = ParseEncoding(env->isolate(), args[1], UTF8);
+
+    if (!StringBytes::IsValidString(args[0].As<String>(), enc)) {
+      return THROW_ERR_INVALID_ARG_VALUE(env, "Invalid encoding for data");
+    }
+
     if (!decoder.Decode(env, args[0].As<String>(), args[1], UTF8)
-             .FromMaybe(false))
+            .FromMaybe(false))
```

Comment:
Unrelated whitespace change?

Reply:
Actually, the new line was removed from the latest commit.